### ISSUE TITLE
c7n - ValueFilter - add value_type: number

### ISSUE THREAD

**Issue by jomalsan** [cloud-custodian/cloud-custodian#4744]
I am working on a policy that returns a version in the format '1.X' and need to check if the version is less than 1.2. Currently the only `value_type` option for converting to numeric is `integer` which floors all of the version numbers to 1 so that they all pass the filter.

Feature request is for a new value_type option, `number` or `float`, that will support converting strings to floating point numbers.

**Comment by thisisshi** [comment]
A `semver` value_type could probably work here,

https://github.com/pypa/packaging

```python
from packaging import version

low = version.parse('0.1.0')
high = version.parse('0.1.1')

assert low < high
```

**Comment by stefangordon** [comment]
Yeah it might make sense to just add both a number/float and a semver.
